examples/restore_login.py

```diff
@@ -1,71 +1,90 @@
+#!/usr/bin/env python3
+
 import asyncio
 import json
 import os
 import sys
+import getpass
 
 from nio import AsyncClient, LoginResponse
 
 CONFIG_FILE = "restore_login_config.json"
 
 # Check out main() below to see how it's done.
 
-def write_details_to_disk(resp: LoginResponse) -> None:
+
+def write_details_to_disk(resp: LoginResponse, home_server) -> None:
     """Writes the required login details to disk so we can log in later without
     using a password.
 
     Arguments:
         resp {LoginResponse} -- the successful client login response.
+        home_server -- the name of the homeserver, e.g. "https://matrix.example.org"
     """
     # open the config file in write-mode
     with open(CONFIG_FILE, "w") as f:
         # write the login details to disk
         json.dump(
             {
-            "access_token": resp.access_token,
-            "device_id": resp.device_id,
-            "user_id": resp.user_id
+                "home_server": home_server,  # e.g. "https://matrix.example.org"
+                "access_token": resp.access_token,  # long cryptographic access token
+                "device_id": resp.device_id,  # device ID, 10 uppercase letters
+                "user_id": resp.user_id  # e.g. "@user:example.org"
             },
             f
         )
 
 
 async def main() -> None:
-    client = AsyncClient("https://matrix.example.org", "@alice:example.org")
 
     # If there are no previously-saved credentials, we'll use the password
     if not os.path.exists(CONFIG_FILE):
-        resp = await client.login("hunter2")
+        print("First time use. Did not find credential file. Asking for homeserver, user, and password to create credential file.")
+        home_server = "https://matrix.example.org"
+        home_server = input(f"Enter your homeserver: [{home_server}] ")
+        if not (home_server.startswith("https://")
+                or home_server.startswith("https://")):
+            home_server = "https://" + home_server
```

Comment:
I'm not sure about this block, it looks like it's testing the same thing twice? If I'm missing something, maybe add a comment.

Reply:
You caught a bug! Thank you. It should be https and http

Fixed in commit 2